src/main/java/com/mg105/use_cases/BattleInteractor.java

```diff
@@ -0,0 +1,299 @@
+package com.mg105.use_cases;
+
+import com.mg105.entities.Battle;
+import com.mg105.entities.BattleCharacter;
+import com.mg105.entities.GameState;
+import com.mg105.entities.Move;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * This class interacts with the Battle entity.
+ * <p>
+ * It is responsible for modifying Battle related entities and for the processing needed during combat encounters.
+ */
+public class BattleInteractor {
+
+    private final GameState state;
+    private Battle encounter;
+
+    private BattlePresenterInterface presenter;
+
+    public BattleInteractor(GameState state) {
+        this.state = state;
+        this.encounter = state.getCurrEncounter();
+    }
+
+    /**
+     * Sets the presenter attribute.
+     *
+     * @param presenter the BattlePresenterInterface instance to set the presenter attribute to.
+     */
+    public void setPresenter(BattlePresenterInterface presenter) {
+        this.presenter = presenter;
+    }
+
+    /**
+     * Creates a new encounter with random opponents and sets it as the current encounter in GameState
+     */
+    public void createEncounter() {
```

Comment:
If you have time you could use a builder pattern here to create the encounter but if you don't just mention it as something that we would do if we had more time (in a comment or something)